### PR TITLE
add ini option to disable string escape for parametrization

### LINKED ISSUE #2482: Is there anyway to not escape id in parametrization?
Thanks for submitting an issue!

Here's a quick checklist in what to include:

- [x] Include a detailed description of the bug or suggestion
- [ ] `pip list` of the virtual environment you are using
- [x] pytest and operating system versions
- [x] Minimal example if possible

-----

I want to print original unicode text without escaping, but I can't find anyway to do that.

code:
```python
import pytest

@pytest.mark.parametrize('text,result', [
    ['테스트1', False],
    ['테스트2', False],
])
def test_func(text):
    assert False
```

result of pytest run
```
$ pytest t.py
======= test session starts =========
platform darwin -- Python 3.6.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- ~/.pyenv/versions/3.6.1/bin/python
cachedir: .cache
plugins: asyncio-0.5.0
collected 2 items

t.py::test_func[\ud14c\uc2a4\ud2b81-False] FAILED
t.py::test_func[\ud14c\uc2a4\ud2b82-False] FAILED
```

Expected result:
```
t.py::test_func[테스트1-False] FAILED
t.py::test_func[테스트2-False] FAILED
```

Unfortunately, I found you are regarding this as a feature in: https://github.com/pytest-dev/pytest/pull/1470

So, for now, I am doing that by adding:
```
import _pytest
_pytest.python._escape_strings = lambda x: x
```

I really need to see the original unicode text, because escaped characters are not human readable. At least, it is a lot better to be displayed in both escaped and unescaped way, not just in escaped way.

Is there any better way?

### PR DESCRIPTION
**Pull request by youknowone**
closes #2482

Cherry-picked and rebased #2830

- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [ ] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [x] Target the `features` branch for new features and removals/deprecations.
- [x] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [ ] Add yourself to `AUTHORS` in alphabetical order;

cc @ApaDoctor

### CODE REVIEW

File: src/_pytest/python.py [hunk 1]
```diff
@@ -1177,7 +1194,7 @@ def _idval(val, argname, idx, idfn, item, config):
             return hook_id
 
     if isinstance(val, STRING_TYPES):
-        return ascii_escaped(val)
+        return _ascii_escaped_by_config(val)
```
Comment (RonnyPfannschmidt) [inline src/_pytest/python.py]: without passing config, this is useless

Comment (youknowone) [reply src/_pytest/python.py]: Thanks, I fixed it

File: doc/en/parametrize.rst [hunk 2]
```diff
@@ -81,6 +81,21 @@ them in turn:
     test_expectation.py:8: AssertionError
     ==================== 1 failed, 2 passed in 0.12 seconds ====================
 
+.. note::
+
+    pytest by default escapes any non-ascii characters used in unicode strings
+    for the parametrization because it has several downsides.
+    If however you would like to use unicode strings in parametrization, use this option in your ``pytest.ini``:
```
Comment (nicoddemus) [inline doc/en/parametrize.rst]:
```suggestion
    If however you would like to use unicode strings in parametrization and see them in the terminal as is (non-escaped), use this option in your ``pytest.ini``:
```

File: doc/en/parametrize.rst [hunk 3]
```diff
@@ -81,6 +81,21 @@ them in turn:
...
+
+    .. code-block:: ini
+
+        [pytest]
+        disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
+
+    to disable this behavior, but keep in mind that this might cause unwanted side effects and
```
Comment (nicoddemus) [inline doc/en/parametrize.rst]:
```suggestion
    Keep in mind however that this might cause unwanted side effects and
```

**Comment by nicoddemus** [comment]
Thanks a lot for this @youknowone!

**Pull request merged**